js/linked-list.js

```diff
@@ -0,0 +1,99 @@
+"use strict";
+
+class Node {
+    constructor(data, next=null){
+        this.data = data;
+        this.next = next;
+    }
+}
+
+class LinkedList {
+    constructor(iterable=null) {
+        this.head = null;
+        this._counter = 0;
+        if(Array.isArray(iterable)){
+            iterable.map(x => this.push(x));
```

Comment:
Ooh I like this! Nice! A better option would've been `iterable.forEach( x => this.push(x))` because you don't need to return a modified array